`tsgolint` panics if (Windows) path contains non-ASCII characters

We've gradually started adopting `oxlint` at work, but have run into an issue where one of my colleagues cannot run `oxlint --type-aware` due to his home directory being `C\:/Users/TobiasLægreid/`. This issue does not happen when running `typescript-native` directly however (i.e., all our `tsgo -p tsconfig.react.json --noEmit` and so on works fine on their machine).

## Versions used:

```
devDependencies:
oxlint 1.24.0
oxlint-tsgolint 0.2.1
@typescript/native-preview 7.0.0-dev.20251023.1
```

<details>
<summary>stacktrace from panic</summary>

```
 "C:\Program Files\nodejs\pnpm.cmd" run oxlint
> @corp/app@ oxlint C:\Users\TobiasLægreid\dev\app\frontend
> oxlint --type-aware .
panic: regexp: Compile(`(?i)^((C\:/Users/TobiasL\ægreid/dev/app/frontend/packages/react/src(/.+?)?/[^/]*\.test\.ts[^/]*)|(C\:/Users/TobiasL\ægreid/dev/app/frontend/packages/react/src(/.+?)?/[^/]*\.stories\.ts[^/]*)|(C\:/Users/TobiasL\ægreid/dev/app/frontend/packages/react/src(/.+?)?/[^/]*\.mdx))($|/)`): error parsing regexp: invalid escape sequence: `\æ`
goroutine 299 [running]:
regexp.MustCompile({0xc0000b0840, 0x151})
        /opt/hostedtoolcache/go/1.25.0/x64/src/regexp/regexp.go:313 +0xb4
github.com/microsoft/typescript-go/internal/tsoptions.getWildcardDirectories({0xc000500f70, 0x1, 0x70?}, {0xc0000a9840?, 0xc000118900?, 0x70?}, {0x28?, {0xc000118c60?, 0x7ff62e770331?}})
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/tsoptions/wildcarddirectories.go:37 +0xcd
github.com/microsoft/typescript-go/internal/tsoptions.(*ParsedCommandLine).WildcardDirectories.func1()
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/tsoptions/parsedcommandline.go:211 +0x5b
sync.(*Once).doSlow(0x5b?, 0x5b?)
        /opt/hostedtoolcache/go/1.25.0/x64/src/sync/once.go:78 +0xac
sync.(*Once).Do(...)
        /opt/hostedtoolcache/go/1.25.0/x64/src/sync/once.go:69
github.com/microsoft/typescript-go/internal/tsoptions.(*ParsedCommandLine).WildcardDirectories(0xc00064c140)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/tsoptions/parsedcommandline.go:209 +0x3e
github.com/microsoft/typescript-go/internal/project.(*ConfigFileRegistryBuilder).updateRootFilesWatch(0xc00021a510, {0xc000118c60?, 0x5b?}, 0xc000492930)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/project/configfileregistrybuilder.go:174 +0x69
github.com/microsoft/typescript-go/internal/project.(*ConfigFileRegistryBuilder).reloadIfNeeded(0xc00021a510, 0xc000492930, {0xc000118c60, 0x5b}, {0xc0000203c0, 0x5b}, 0x0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/project/configfileregistrybuilder.go:111 +0x157
github.com/microsoft/typescript-go/internal/project.(*ConfigFileRegistryBuilder).acquireConfigForOpenFile.func2(0xc000492930?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/project/configfileregistrybuilder.go:271 +0x10a
github.com/microsoft/typescript-go/internal/project/dirty.(*SyncMapEntry[...]).changeLocked(0xc0004dfcb8?, 0x2?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/project/dirty/syncmap.go:105 +0x1b6
github.com/microsoft/typescript-go/internal/project/dirty.(*SyncMapEntry[...]).ChangeIf(0x7ff62f1ff880, 0xc0000203c0, 0xc0004dfd40?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/project/dirty/syncmap.go:135 +0xc5
github.com/microsoft/typescript-go/internal/project.(*ConfigFileRegistryBuilder).acquireConfigForOpenFile(0xc00021a510, {0xc000118c60, 0x5b}, {0xc0000203c0, 0x5b}, {0xc000490ee0, 0x62}, 0x0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/project/configfileregistrybuilder.go:258 +0x24e
github.com/microsoft/typescript-go/internal/project.(*ConfigFileRegistryBuilder).FindOrAcquireConfigForOpenFile(0xc00021a540?, {0xc000118c60?, 0x5b?}, {0xc0000203c0?, 0xc0004dfe68?}, {0xc000490ee0?, 0xc00037ce68?}, 0x7ff62ec046c0?, 0xc0004dfe48?)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/project/configfileregistrybuilder.go:93 +0x37
github.com/typescript-eslint/tsgolint/internal/utils.(*TsConfigResolver).findConfigWithReferences.func2({{0xc000118c60?, 0xc00026ef60?}})
        /home/runner/work/tsgolint/tsgolint/internal/utils/find_tsconfig.go:90 +0xc9
github.com/typescript-eslint/tsgolint/internal/utils.BreadthFirstSearchParallelEx[...].func3(0xc0000088d0)
        /home/runner/work/tsgolint/tsgolint/internal/utils/bfs.go:117 +0x128
created by github.com/typescript-eslint/tsgolint/internal/utils.BreadthFirstSearchParallelEx[...].func1-range1 in goroutine 1
        /home/runner/work/tsgolint/tsgolint/internal/utils/bfs.go:102 +0x22f
Error running tsgolint: "exit status: exit code: 2" ELIFECYCLE  Command failed with exit code 1.
Process finished with exit code 1
```

</details>

--- a/patches/0006-fix-panic-when-compiling-wildcard-directories-patter.patch
+++ b/patches/0006-fix-panic-when-compiling-wildcard-directories-patter.patch
@@ -1,0 +1,105 @@
+From 7652de1cf2d626000f562884ad104bc213213ecb Mon Sep 17 00:00:00 2001
+From: Cameron Clark <cameron.clark@hey.com>
+Date: Fri, 24 Oct 2025 11:24:09 +0100
+Subject: [PATCH] fix panic when compiling wildcard directories pattern
+
+---
+ .../tsoptions/wildcarddirectories_test.go     | 65 +++++++++++++++++++
+ internal/vfs/utilities.go                     |  8 +--
+ 2 files changed, 69 insertions(+), 4 deletions(-)
+ create mode 100644 internal/tsoptions/wildcarddirectories_test.go
+
+diff --git a/internal/tsoptions/wildcarddirectories_test.go b/internal/tsoptions/wildcarddirectories_test.go
+new file mode 100644
+index 000000000..65976295a
+--- /dev/null
++++ b/internal/tsoptions/wildcarddirectories_test.go
+@@ -0,0 +1,65 @@
++package tsoptions
++
++import (
++	"testing"
++
++	"github.com/microsoft/typescript-go/internal/tspath"
++)
++
++func TestGetWildcardDirectories_NonASCIICharacters(t *testing.T) {
++	t.Parallel()
++
++	tests := []struct {
++		name                      string
++		include                   []string
++		exclude                   []string
++		currentDirectory          string
++		useCaseSensitiveFileNames bool
++	}{
++		{
++			name:                      "Norwegian character æ in path",
++			include:                   []string{"src/**/*.test.ts", "src/**/*.stories.ts", "src/**/*.mdx"},
++			exclude:                   []string{"node_modules"},
++			currentDirectory:          "C:/Users/TobiasLægreid/dev/app/frontend/packages/react",
++			useCaseSensitiveFileNames: false,
++		},
++		{
++			name:                      "Japanese characters in path",
++			include:                   []string{"src/**/*.ts"},
++			exclude:                   []string{"テスト"},
++			currentDirectory:          "/Users/ユーザー/プロジェクト",
++			useCaseSensitiveFileNames: true,
++		},
++		{
++			name:                      "Chinese characters in path",
++			include:                   []string{"源代码/**/*.js"},
++			exclude:                   []string{"节点模块"},
++			currentDirectory:          "/home/用户/项目",
++			useCaseSensitiveFileNames: true,
++		},
++		{
++			name:                      "Various Unicode characters",
++			include:                   []string{"src/**/*.ts"},
++			exclude:                   []string{"node_modules"},
++			currentDirectory:          "/Users/Müller/café/naïve/résumé",
++			useCaseSensitiveFileNames: false,
++		},
++	}
++
++	for _, tt := range tests {
++		t.Run(tt.name, func(t *testing.T) {
++			t.Parallel()
++
++			comparePathsOptions := tspath.ComparePathsOptions{
++				CurrentDirectory:          tt.currentDirectory,
++				UseCaseSensitiveFileNames: tt.useCaseSensitiveFileNames,
++			}
++
++			result := getWildcardDirectories(tt.include, tt.exclude, comparePathsOptions)
++
++			if result == nil {
++				t.Fatalf("expected non-nil result")
++			}
++		})
++	}
++}
+diff --git a/internal/vfs/utilities.go b/internal/vfs/utilities.go
+index cf0e0e118..5b9d1ff25 100644
+--- a/internal/vfs/utilities.go
++++ b/internal/vfs/utilities.go
+@@ -83,11 +83,11 @@ func IsImplicitGlob(lastPathComponent string) bool {
+ 	return !strings.ContainsAny(lastPathComponent, ".*?")
+ }
+ 
+-// Reserved characters, forces escaping of any non-word (or digit), non-whitespace character.
+-// It may be inefficient (we could just match (/[-[\]{}()*+?.,\\^$|#\s]/g), but this is future
+-// proof.
++// Reserved characters - only escape actual regex metacharacters.
++// Go's regexp doesn't support \x escape sequences for arbitrary characters,
++// so we only escape characters that have special meaning in regex.
+ var (
+-	reservedCharacterPattern *regexp.Regexp = regexp.MustCompile(`[^\w\s/]`)
++	reservedCharacterPattern *regexp.Regexp = regexp.MustCompile(`[\\.\+*?()\[\]{}^$|#]`)
+ 	wildcardCharCodes                       = []rune{'*', '?'}
+ )
+ 
+-- 
+2.50.1 (Apple Git-155)
+

